paparazzi/src/test/java/app/cash/paparazzi/internal/apng/APNGReaderTest.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.cash.paparazzi.internal.apng
+
+import app.cash.paparazzi.internal.ImageUtils
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import java.io.File
+import javax.imageio.ImageIO
+
+class APNGReaderTest {
+
+  @Test
+  fun decodesAllFrames() {
+    val file = javaClass.classLoader.getResource("simple_animation.png")
+    val reader = APNGReader(File(file!!.toURI()))
+
+    for (i in 0 until 3) {
+      val expectedFile = javaClass.classLoader.getResource("simple_animation_$i.png")
+      val expectedImage = ImageIO.read(expectedFile)
+      val actualImage = reader.getNextFrame()!!
+      ImageUtils.assertImageSimilar(expectedFile!!.path, expectedImage, actualImage, 0.0)
+    }
+
+    assertThat(reader.finished()).isTrue()
+    assertThat(reader.frameCount).isEqualTo(reader.frameNumber)
+  }
+
+  @Test(expected = IllegalStateException::class)
```

Comment:
IIRC, technically `@Test(expected)` is considered bad form, since the exception doesn't specify precisely the call site from which it's thrown.

Accordingly, can we move this into the form:

```
try {
  body()
  fail()
} catch (e: IllegalStateException) {
  assertThat(e).hasMessage("whevs")
} 
```
?

Reply:
The `Test` annotation Javadoc explains it, and there's a org.junit.Assert.assertThrows for replacement rather than `try-catch`. There's also a Truth's `ExpectFailure`, but I can't decipher how to use it in a simple way.